tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -375,3 +375,25 @@ def validate_header_section_matches_submission(datafile, section, generate_error
         )
 
     return is_valid, error
+
+def validate_header_tribe_code(program_type, tribe_code, state_fips_code, generate_error):
+    """Validate header tribe code, fips code, and program type all agree with eachother."""
+    is_valid = False
+
+    if program_type == 'TAN' and (state_fips_code == '00' or state_fips_code is None or state_fips_code.isspace()):
+        is_valid = tribe_code is not None and not tribe_code.isspace() and tribe_code != '000'
+    else:
+        is_valid = tribe_code is None or tribe_code.isspace() or tribe_code == '000'
```

Comment:
These expressions seem very very close and almost the same check is being done against the `state_fips_code` in the conditional. I think refactoring here could lead to increased readability.